### PR TITLE
refactor(hx-div): compatability audit updates

### PR DESCRIPTION
**Pull request by 100stacks**
* refactor `HXDiv` based on `Compatibility Audit`

JIRA: SURF-1516

### LGTM's
- [x] Dev LGTM
- [x] Zoom LGTM

### CODE REVIEW

**Comment by HelixUI** [comment]
Deploy preview for *helix-ui* processing.

Built with commit bfe30dc7904b8b58db6dd0f1f3893fbe9e1c5d89

https://app.netlify.com/sites/helix-ui/deploys/5c3557fdf6198f000860fee6

File: src/helix-ui/elements/HXDivElement.js [hunk 1]
```diff
@@ -33,22 +33,21 @@ export class HXDivElement extends HXElement {
...
-                // FIXME: listener doesn't get removed if element disconnects
                 this.removeEventListener('scroll', onScroll);
             }
         }
     }
 
+    $onDisconnect () {
+        this.removeEventListener('scroll', onScroll);
+    }
+
```
Comment (100stacks) [inline src/helix-ui/elements/HXDivElement.js]: Added `$onDisconnect()` based on the comment:

> // FIXME: listener doesn't get removed if element disconnects

https://github.com/rackerlabs/helix-ui/pull/418/files#diff-f7d3735db68684c95b1432479629fe2aL46

I didn't noticed any patterns without the `$onConnect()`, so adding a note here to verify this with the team.

Comment (CITguy) [reply src/helix-ui/elements/HXDivElement.js]:
```javascript
$onConnect () {
  ...
  if (this.hasAttribute('scroll')) {
    this._addScrollListeners();
  }
}
```

File: src/helix-ui/elements/HXDivElement.js [hunk 2]
```diff
@@ -33,22 +33,21 @@ export class HXDivElement extends HXElement {
         return [ 'scroll' ];
     }
 
-    // FIXME: ensure 'scroll' event listener added on connect (if scroll attr present)
-    // FIXME: ensure 'scroll' event listener is removed on disconnect
-
     $onAttributeChange (attr, oldVal, newVal) {
-        if (attr === 'scroll') {
-            // FIXME: only run if connected
+        if (this.isConnected && attr === 'scroll') {
```
Comment (CITguy) [inline src/helix-ui/elements/HXDivElement.js]: Please remove `this.isConnected` from conditions.

**Pull request merged**